syntax = "proto3";

message LockRequest {
    string key = 1;
}

message LockResponse {
    bool lock_acquired = 1;
}

message Request {
    string key = 1;
    string value = 2;
}

message Response {
    string message = 1;
}

service KeyValueStore {
    rpc Put (Request) returns (Response);
    rpc Get (Request) returns (Response);
    rpc Delete (Request) returns (Response);
    rpc AcquireLock (LockRequest) returns (LockResponse);
    rpc ReleaseLock (LockRequest) returns (LockResponse);
}

